dist/tools/jlink/jlink.sh

```diff
@@ -0,0 +1,214 @@
+#               hexfile is expected in ihex format and is pointed to
+#               by BINFILE environment variable
+#
+#               options:
+#               BINFILE: path to the binary file that is flashed
+#
+# debug:        starts OpenOCD as GDB server in the background and
+#               connects to the server with the GDB client specified by
+#               the board (DBG environment variable)
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TELNET_PORT:    port opened for telnet connections
+#               TUI:            if TUI!=null, the -tui option will be used
+#               ELFFILE:        path to the ELF file to debug
+#
+# debug-server: starts OpenOCD as GDB server, but does not connect to
+#               to it with any frontend. This might be useful when using
+#               IDEs.
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TELNET_PORT:    port opened for telnet connections
+#
+# reset:        triggers a hardware reset of the target board
+#
+#
+# @author       Hauke Peteresen <hauke.petersen@fu-berlin.de>
+
+# default GDB port
+_GDB_PORT=3333
+# default telnet port
+_TELNET_PORT=4444
+# default JLink command, interface and speed
+_JLINK=JLinkExe
+_JLINK_SERVER=JLinkGDBServer
+_JLINK_IF=SWD
+_JLINK_SPEED=2000
+# default starting address of the devices flash memory
+_FLASH_ADDR=0
+
+#
+# a couple of tests for certain configuration options
+#
+test_config() {
+
+    if [ -z "${HEXFILE}" ]; then
+        echo "no hexfile"
+    else
+        echo "HEXFILE found"
+    fi
+
+    if [ -z "${JLINK}" ]; then
+        JLINK=${_JLINK}
+    fi
+    if [ -z "${JLINK_SERVER}" ]; then
+        JLINK_SERVER=${_JLINK_SERVER}
+    fi
+    if [ -z "${JLINK_IF}" ]; then
+        JLINK_IF=${_JLINK_IF}
+    fi
+    if [ -z "${JLINK_SPEED}" ]; then
+        JLINK_SPEED=${_JLINK_SPEED}
+    fi
+    if [ -z "${JLINK_DEVICE}" ]; then
+        echo "Error: No target device defined in JLINK_DEVICE env var"
+        exit 1
+    fi
+    if [ -z "${JLINK_FLASH_ADDR}" ]; then
+        JLINK_FLASH_ADDR=${_FLASH_ADDR}
+    fi
+}
+
+test_hexfile() {
+    if [ ! -f "${HEXFILE}" ]; then
+        echo "Error: Unable to locate HEXFILE"
+        echo "       (${HEXFILE})"
+        exit 1
+    fi
+}
+
+test_ports() {
+    if [ -z "${GDB_PORT}" ]; then
+        GDB_PORT=${_GDB_PORT}
+    fi
+    if [ -z "${TELNET_PORT}" ]; then
+        TELNET_PORT=${_TELNET_PORT}
+    fi
+}
+
+test_elffile() {
+    if [ ! -f "${ELFFILE}" ]; then
+        echo "Error: Unable to locate ELFFILE"
+        echo "       (${ELFFILE})"
+        exit 1
+    fi
+}
+
+test_tui() {
+    if [ -n "${TUI}" ]; then
+        TUI=-tui
+    fi
+}
+
+#
+# now comes the actual actions
+#
+do_flash() {
+    test_config
+    test_hexfile
+    # clear any existing contents in burn file
+    echo -n "" > ${BINDIR}/burn.seg
```

Comment:
Please use `/bin/echo` in jlink.sh. `sh`'s builtin echo behaves strange on OS X.
```sh
sh-3.2$ echo -n hello
-n hello
sh-3.2$ /bin/echo -n hello
hellosh-3.2$
```